### PR TITLE
Split OTLP receiver by protocols, allow mTLS support

### PR DESCRIPTION
**Pull request by bogdandrutu**
**Description:** <Describe what has changed. 
The `github.com/soheilhy/cmux` does not support mTLS, so we have to split the protocols and avoid using any `mux` to allow support for mTLS.

Temporary remove the support for `unix` sockets, will be added back soon with a change to grpc server settings to support transport.

### CODE REVIEW

File: receiver/otlpreceiver/testdata/config.yaml [hunk 1]
```diff
@@ -2,51 +2,72 @@ receivers:
   # The following entry initializes the default OTLP receiver.
   # The full name of this receiver is `otlp` and can be referenced in pipelines by 'otlp'.
   otlp:
+    protocols:
+      grpc:
+      http:
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/testdata/config.yaml:7]: These are required to enable the protocol, right? If any of these lines is missing that particular protocol will not be enabled. Do I understand it correctly?

File: receiver/otlpreceiver/testdata/config.yaml [hunk 2]
```diff
@@ -2,51 +2,72 @@ receivers:
...
-    transport: unix
-    endpoint: /tmp/otlp.sock
-
+    protocols:
+      grpc:
+        # transport: unix
+        endpoint: /tmp/grpc_otlp.sock
+      http:
+        # transport: unix
+        endpoint: /tmp/http_otlp.sock
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/testdata/config.yaml:62]: PR description says Unix socket is not supported. In that case it is better to remove this from the config.

File: receiver/otlpreceiver/testdata/config.yaml [hunk 3]
```diff
@@ -2,51 +2,72 @@ receivers:
...
-        max_connection_idle: 11s
-        max_connection_age: 12s
-        max_connection_age_grace: 13s
-        time: 30s
-        timeout: 5s
-      enforcement_policy:
-        min_time: 10s
-        permit_without_stream: true
+    protocols:
+      grpc:
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/testdata/config.yaml:20]: Is grpc-gateway enabled by "grpc" protocol or by "http" protocol?

File: receiver/otlpreceiver/otlp.go [hunk 4]
```diff
@@ -17,255 +17,140 @@ package otlpreceiver
...
 	}
-
-	for _, opt := range opts {
-		opt.withReceiver(r)
+	if cfg.GRPC != nil {
+		opts, err := cfg.GRPC.ToServerOption()
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, grpc.ReadBufferSize(512*1024))
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/otlp.go:61]: Rebase and use the [new config option](https://github.com/open-telemetry/opentelemetry-collector/blob/7ef6c533d4079e9d640bb7d7ac7b61e999984d4e/config/configgrpc/configgrpc.go#L76) for buffer size. :-)

File: receiver/otlpreceiver/testdata/config.yaml (hunk 2 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/testdata/config.yaml:62]: I am planning to add it back in the next couple of PRs, and want to just uncomment that when it happens.

File: receiver/otlpreceiver/testdata/config.yaml (hunk 3 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/testdata/config.yaml:20]: By http.

File: receiver/otlpreceiver/otlp.go (hunk 4 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp.go]: Done.

**Comment by bogdandrutu** [comment]
> Please also update otlpreceiver/README.md
Done.

File: receiver/otlpreceiver/testdata/config.yaml (hunk 1 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/testdata/config.yaml:7]: Commented in the README about this.

**Comment by jmacd** [comment]
Can we reconsider this change? This is a huge regression for usability.

The fact that github.com/soheilhy/cmux does not support mTLS on a shared port does not sound like a convincing reason to require multiple configurations for gRPC vs HTTP receivers.

**Pull request merged**